framework/src/utils/MooseEnumItem.C

```diff
@@ -0,0 +1,68 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "MooseEnumItem.h"
+#include "MooseUtils.h"
+
+MooseEnumItem::MooseEnumItem(const std::string & name, const int & id)
+  : _raw_name(MooseUtils::trim(name)),
+    _name(MooseUtils::toUpper(_raw_name)),
+    _id(id),
+    _unique(std::hash<std::string>()(std::to_string(_id) + _name))
```

Comment:
You might join this with a separator character to avoid getting two identical strings sent to the hasher. Yes this would be highly unlikely but it would be possible.  (1 + 0foo, and 10 + foo).